File: src/design_patterns/creational/builder/php/pizza/HawaiianPizzaBuilder.php
<?php

class HawaiianPizzaBuilder extends PizzaBuilder {

    public function buildDough() {
        $this->pizza->setDough("cross");
    }

    public function buildSauce() {
        $this->pizza->setSauce("mild");
    }

    public function buildTopping() {
        $this->pizza->setTopping("ham + pineapple");
    }
}

File: src/design_patterns/creational/builder/java/Pizza/HawaiianPizzaBuilder.java
/* "ConcreteBuilder" */
class HawaiianPizzaBuilder extends PizzaBuilder {
	public void buildDough()   { pizza.setDough("cross"); }
	public void buildSauce()   { pizza.setSauce("mild"); }
	public void buildTopping() { pizza.setTopping("ham+pineapple"); }
}

